vectordatasource/transform.py

```diff
@@ -3618,9 +3628,19 @@ def merge_polygon_features(ctx):
     if end_zoom is not None and zoom >= end_zoom:
         return None
 
+    tfz = tolerance_for_zoom(zoom)
+    if buffer_merge_tolerance:
+        tolerance = eval(buffer_merge_tolerance, {}, {
+            'tolerance_for_zoom': tfz,
+        })
+    else:
+        tolerance = tfz
+
     def _props_pre((shape, props, fid)):
         # drop area while merging, as we'll recalculate after.
         props.pop('area', None)
+        if merge_min_zooms:
+            props.pop('min_zoom', None)
```

Comment:
What is the `min_zoom` of the resulting polygon, I'm reading this as it's stripped off? 

Could it be the min of the input `min_zoom` values instead?

Reply:
(Perhaps this is answered in the new `_props_merge` function below... but can you walk thru an example, please?

Reply:
Yup, that's exactly what's happening. When `_merge_features_by_property` runs, it calls our `_props_pre` function for each feature, and whatever we return is the "key" used to group features to be merged. We `pop` the `min_zoom` off the properties so that the merge algorithm will merge adjacent polygons with different `min_zoom` - which happens all the time in the patchwork of highly variable-sized fields in the 3dshapes import.

However, `_merge_features_by_property` (now) also keeps the original properties, and passes the list of those into our [`_props_merge` function](https://github.com/tilezen/vector-datasource/pull/1791/files/268f1e8d661ffa7d13b7245986c8f4804db4ab7d#diff-ccacdffc74e3f0be743bf4d7d6a1bb9cR3652) below, and the merged feature gets assigned whatever we return. The `_props_merge` implementation I've used here takes the minimum of all the `min_zoom`s, as this will guarantee that we see the merged feature at least at all the zooms of the biggest feature we queried from the database or RAWR tile.

The step-by-step run-through might be something like:

1. We call `merge_polygon_features` on a layer with two features, `[(shape1, props1, fid1), (shape2, props2, fid2)]`.
2. `merge_polygon_features` calls `_merge_features_by_property`.
3. `_merge_features_by_property` calls back into `merge_polygon_features`' `_props_pre` for each feature, where we strip the min zoom. In this example, `props1.pop('min_zoom') == props2.pop('min_zoom') == props`.
4. `_merge_features_by_property` stores the features indexed by that result, so we get something like: `{props: [(shape1, props1), (shape2, props2)]}`
5. `_merge_features_by_property` tries to merge `shape1` and `shape2`, getting `shape_merged`
6. `_merge_features_by_property` calls back into `merge_polygon_features`' `_props_merge` with `[props1, props2]`, and we take the min of all the `min_zoom` parameters, returning `props_merged`.
7. `_merge_features_by_property` calls back into `merge_polygon_features`' `_props_post`, where it resets the `area`  on `props_merged` (and is why we don't bother with that in step 6).
8. `_merge_features_by_property` returns a new list of features: `[(shape_merged, props_merged, some_fid)]`, where `some_fid` is one of the `fids` of one of the merged features.